### DISCUSSION TITLE
Converting EPS to JPG using magickload (ImageMagick 7) returns an error

### DISCUSSION THREAD

**Discussion by ariefatkhur** [libvips/libvips#2765] [Q&A] [answered]
## Questions, enhancements, tips, etc.

Please use libvips discussions for other topics.

https://github.com/libvips/libvips/discussions

## Bug report

**Describe the bug**
Converting EPS to JPG using magickload returns an error. Standalone installation of ImageMagick 7.1.0-29 does this perfectly.

**To Reproduce**
1. Use eps file saved from Adobe Photoshop 23.2 (Macintosh)
2. Use libvips built against ImageMagick 7.1.0-29 for Windows
3. Run `vips.exe magickload input.eps output.jpg`
4. See error: something like "gs command is not recognized"
5. Install AGPL ghostscript
6. Rerun step 3
7. Error: `magickload: Magick: NoDecodeDelegateForThisImageFormat '' @ error/constitute.c/ReadImage/737 (null)`

**Expected behavior**
Outputs 2 files, the main image (output-0.jpg) and a thumbnail (output-1.jpg). Standalone installation of ImageMagick 7 does this. Command used (with legacy command support) : `convert.exe input.eps output.jpg`

**Actual behavior**
Error: `magickload: Magick: NoDecodeDelegateForThisImageFormat '' @ error/constitute.c/ReadImage/737 (null)`

**Environment**
 - OS: Windows 10
 - Vips: 8.12 built using [build-win64-mxe](https://github.com/libvips/build-win64-mxe) and ImageMagick 7.1.0-29, built on Ubuntu 20.04.4 LTS 2 Virtual Machine.

**Comment by jcupitt** [comment, marked as answer]
Hi @swdmnd,

The libvips windows binaries only include the imagemagick loaders which are actually built into imagemagick. Imagemagick doesn't decode EPS itself, it delegates decode to GhostScript. In the same way, IM doesn't decode SVG, it delegates to Inkscape.

This is partly for size reasons (the download would be huge if we included all these other packages as well), but also partly for licensing reasons. GhostScript is licenced under the AGPL, and we would not be allowed to distribute it. Moreover, GS cannot be used in closed-source software, so it would prevent any commercial use of libvips.

You can decode EPS with libvips under WSL2 on Windows:

```
$ vipsheader -a hayai\ logo.eps 
hayai logo.eps: 176x222 uchar, 4 bands, srgb, magickload
width: 176
height: 222
bands: 4
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 1
yres: 1
filename: hayai logo.eps
vips-loader: magickload
icc-profile-data: 2576 bytes of binary data
magick-date:create: 2022-04-15T09:27:24+00:00
magick-date:modify: 2022-04-15T09:27:24+00:00
magick-icc:copyright: Copyright Artifex Software 2011
magick-icc:description: Artifex Software sRGB ICC Profile
magick-ps:HiResBoundingBox: 176x222+214+108
magick-ps:Level: PS-Adobe-3.0 EPSF-3.0
n-pages: 1
```

But of course you can't decode EPS using imagemagick in any commercial software, so you would need to be very cautious.

**Reply by ariefatkhur** [reply]
Thank you for the explanation.

However, I'm still confused that ImageMagick 7 standalone installation can do eps conversion without any ghostscript installed on my computer but not on my friend's computer (it's asking for gswin64c). Do you have any idea why?

**Reply by jcupitt** [reply]
I think installing imagemagick on windows will install a private copy ghostscript and all that stuff -- that's why the download is so large. It has a series of mysterious registry settings that the various components use to find each other.
